kubeai/README.md

```diff
@@ -181,6 +181,109 @@ curl "http://localhost:8000/openai/v1/chat/completions" \
 
 Enjoy the answer!
 
+# CPU Performance Optimization with NRI
+
+## Overview
+
+[NRI plugins][nri-plugins] provide a way to
+optimize the resource placement of applications in a Kubernetes cluster. They
+connect to the container runtime and are able, for example, to adjust the CPU
+and memory pinning of containers.
+
+This section provides a guide on how to use the
+[Balloons Policy][balloons-policy] plugin from the [NRI Plugins][nri-plugins]
+project to optimize the performance of CPU-backed KubeAI profiles.
+
+## Installation of Balloons Policy Plugin
+
+> **NOTE:** In order to prevent disturbing already running workloads it is
+> recommended to install the NRI plugin to an empty node (right after node
+> bootstrap or drain the node before installation).
+
+Install the balloons policy plugin with Helm:
+
+```bash
+helm repo add nri-plugins https://containers.github.io/nri-plugins
+helm repo update nri-plugins
+helm install -n kube-system balloons nri-plugins/nri-resource-policy-balloons
+```
+
+> **NOTE**: If using containerd version earlier than v2.0 you need to enable
+> the NRI support in the containerd configuration file. Instead of manual
+> configuration you can provide `--set nri.runtime.patchConfig=true` to the Helm
+> command above, which will automatically patch the containerd configuration
+> file on each node.
+
+Verify that the balloons policy plugin is running on every node:
+
+```bash
+$ kubectl -n kube-system get ds nri-resource-policy-balloons
+NAME                           DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
+nri-resource-policy-balloons   2         2         2       2            2           kubernetes.io/os=linux   77s
+```
+
+## Configuration of Balloons Policy Plugin
+
+The aim of the balloons policy configuration is to isolate the vllm containers
+to minimize their interference with other containers.
+
+An example configuration for the current CPU-backed model profiles:
+
+```yaml
+cat <<EOF | kubectl apply -f -
+apiVersion: config.nri/v1alpha1
+kind: BalloonsPolicy
+metadata:
+  name: default
+  namespace: kube-system
+spec:
+  allocatorTopologyBalancing: true
+  balloonTypes:
+  - name: kubeai-vllm
+    allocatorPriority: high
+    minBalloons: 1
+    minCPUs: 12
+    preferNewBalloons: true
+    hideHyperthreads: true
+    matchExpressions:
+    - key: labels/app.kubernetes.io/name
+      operator: In
+      values: ["vllm"]
+    - key: name
+      operator: In
+      values: ["server"]
+  - name: default
+    namespaces:
+    - "*"
+    shareIdleCPUsInSame: numa
+  pinCPU: true
+  pinMemory: false
+  reservedPoolNamespaces:
+  - kube-system
+  reservedResources:
+    cpu: "2"
+  log:
+    debug: ["policy"]
+  instrumentation:
+    httpEndpoint: :8891
+    prometheusExport: true
+EOF
+```
+
+The configuration above allocates at minimum 12 logical CPUs (6 full CPU cores
+assuming that hyperthreading is enabled) for the inference serving pods. With
```

Comment:
Pods, not containers?
```suggestion
assuming that hyperthreading is enabled) for the vLLM pods. With
```

Reply:
I heavily rephrased this paragraph. No reference to vLLM anymore (just "inference")